Either urh works with numpy >= 2.0, or it has been long broken anyway.

--- a/setup.py
+++ b/setup.py
@@ -142,7 +142,7 @@
         return ""
 
 
-install_requires = ["numpy<2.0.0", "psutil", "cython", "setuptools"]
+install_requires = ["numpy", "psutil", "cython", "setuptools"]
 if IS_RELEASE:
     install_requires.append("pyqt5")
 else:
@@ -168,7 +168,7 @@
     license="GNU General Public License (GPL)",
     download_url="https://github.com/jopohl/urh/tarball/v" + str(version.VERSION),
     install_requires=install_requires,
-    setup_requires=["numpy<2.0.0"],
+    setup_requires=["numpy"],
     packages=get_packages(),
     ext_modules=get_extensions(),
     cmdclass={"build_ext": build_ext},
